### PR TITLE
amp-bind: Support amp-youtube

### PR DESCRIPTION
**Pull request by choumx**
Partial for #7825.

- Support binding to `amp-youtube`'s `data-videoid` attribute.

/to @aghassemi

### CODE REVIEW

File: extensions/amp-youtube/0.1/amp-youtube.js [hunk 1]
```diff
@@ -211,6 +205,28 @@ class AmpYoutube extends AMP.BaseElement {
     }
   }
 
+  /** @override */
+  mutatedAttributesCallback(mutations) {
+    if (mutations['data-videoid'] !== undefined) {
+      this.videoid_ = this.getVideoId_();
+      if (this.iframe_) { // `null` if element hasn't been laid out yet.
```
Comment (aghassemi) [inline extensions/amp-youtube/0.1/amp-youtube.js]: no need to refresh the iframe, YouTube supports dynamically loading a video which should be much faster than a full reload, the following should work (not tested though)

```
    this.playerReadyPromise_.then(() => {
      this.sendCommand_('loadVideoById', this.videoid_);
    });
```

Comment (choumx) [inline extensions/amp-youtube/0.1/amp-youtube.js]: Nice, thanks for pointing this out.

Comment (aghassemi) [inline extensions/amp-youtube/0.1/amp-youtube.js]: There is short amount of time (actually a good 2 seconds, so not that short :) ) when this.iframe_ is not null but player is not loaded either (meaning commands will be ignored) wrapping the `sendCommand_` in the `this.playerReadyPromise_.then` will queue it up in those cases.

Comment (choumx) [inline extensions/amp-youtube/0.1/amp-youtube.js]: I actually moved the `this.playerReadyPromise_.then` to within `sendCommand_` to reduce duplication. I assume that we never want to call `sendCommand_` before the player is ready as you mentioned.

**Comment by choumx** [comment]
> Can you please add a new test for this on the bind integration tests?

Done, though the added value isn't great since it doesn't test any logic within `amp-youtube`. That would probably require [`getVideoUrl`](https://developers.google.com/youtube/iframe_api_reference#Retrieving_video_information) YT API.

File: extensions/amp-youtube/0.1/amp-youtube.js (hunk 1 above)
Comment (aghassemi) [inline extensions/amp-youtube/0.1/amp-youtube.js]: :+1:

**Comment by choumx** [comment]
@kmh287 Gentle ping.

**Comment by choumx** [comment]
> Just curious: do we need to also update the bind-validator to specify allowed protocols for amp-youtube like we do for amp-video?

I don't think so since `amp-brightcove.js` constructs the URL from fragments like "account ID".

**Pull request merged**